init.c

```diff
@@ -506,11 +506,14 @@ static int add_to_spam_list (SPAM_LIST **list, const char *pat, const char *temp
   {
     t = mutt_new_spam_list();
     t->rx = rx;
+    rx = NULL;
     if (last)
       last->next = t;
     else
       *list = t;
   }
+  else
+    mutt_free_regexp(&rx);
 
   /* Now t is the SPAM_LIST* that we want to modify. It is prepared. */
```

Comment:
You'll need to expand the view above and below, to follow this comment.
- We find an entry `t` in the spam list (`for` loop above)
- We dereference `t->rx` (`if` below) without checking it

This isn't something you've introduced (and it may not even be a problem).